cmd/habitat/community/manager.go

```diff
@@ -199,6 +205,28 @@ func (m *Manager) JoinCommunity(name string, swarmkey string, btstps []string, a
 	return stateMachine.State()
 }
 
+func (m *Manager) AddMemberNode(communityID string, member *community.Member, node *community.Node) (*community.CommunityState, error) {
```

Comment:
Since we decided that members are always tied to nodes, should we encode this in the transitions? like have this be one transition instead of two? just thinking this might simplify the state model because then there shouldn't be partial failures where a node gets added without a member.

like maybe we have `AddMemberWithNode` and `AddMemberWithoutNode` as two specific transitions. lmk what you think

Reply:
I think its fine to have them be separate. The validator function we write for the transition should check if an invalid action is made (like adding a node attached to a non-existent member). In addition, since the last PR added batched transitions, we basically get them in the same raft proposal